src/colinajose/model/schoolEntities/School.java

```diff
@@ -0,0 +1,372 @@
+package colinajose.model.schoolEntities;
+
+import colinajose.data.DataFileIO;
+import colinajose.model.notification.Device;
+import colinajose.model.people.Person;
+import datastructures.arraylist.MyArrayList;
+import datastructures.circulardoublylinkedlist.MyCircularDoublyLinkedList;
+import colinajose.model.base.BaseEntity;
+import colinajose.model.notification.Observable;
+import colinajose.model.notification.Observer;
+import colinajose.model.people.Contact;
+import colinajose.model.people.Employee;
+import colinajose.model.people.Parent;
+import colinajose.model.people.Principal;
+import colinajose.model.people.Secretary;
+import colinajose.model.people.Student;
+import colinajose.model.people.Teacher;
+import datastructures.hashmap.MyHashMap;
+
+public class School extends BaseEntity implements Observable {
```

Comment:
Jose, I know I asked you for this but now seeing the complexity of this class I don't like it. You were right can you please make a School Observable Class or something like that. That will take about the implementation of the methods from Observable Interface.
One other thing I'll like is that on this school class you don't have any logic only define the attributes the getters and setters. Then create a new class but under logic package, not model, called SchoolService or something like that and transfer the actual logic of adding grades, etc.